src/index.js

```diff
@@ -26,40 +27,119 @@ type Options = {
                 urlParams: query,
                 headers,
-                formFields,
+                formFields: {},
             });
 
             if (res) {
-                const {headers: responseHeaders, content, statusCode} = res;
-
-                response.statusCode = statusCode;
-                Object.keys(responseHeaders).forEach((h) => response.setHeader(h, responseHeaders[h]));
-                response.write(content);
-                response.end();
+                return sendResponse(response, res.headers, res.statusCode, res.content);
             }
-        });
+        }
+
+        if (isMultipart(contentType)) {
+            const form = new multiparty.Form();
+            form.parse(request, (err, formFields = {}, files) => {
+                console.log(err, formFields, files);
+                const res = mockingServer.handle({
+                    method,
+                    urlPath: pathname || '',
+                    urlParams: query,
+                    headers,
+                    formFields,
+                });
+
+                if (res) {
+                    const {headers: responseHeaders, content, statusCode} = res;
+
+                    sendResponse(response, responseHeaders, statusCode, content);
+                }
+            });
+        } else if (isFormUrlencoded(contentType)) {
+            let body = '';
+
+            request.on('data', (data) => {
+                body += data;
+            });
+
+            request.on('end', () => {
+                const res = mockingServer.handle({
+                    method,
+                    urlPath: pathname || '',
+                    urlParams: query,
+                    headers,
+                    formFields: querystring.parse(body),
+                });
+
+                if (res) {
+                    const {headers: responseHeaders, content, statusCode} = res;
+
+                    sendResponse(response, responseHeaders, statusCode, content);
+                }
+            });
+        } else {
+            let body = '';
+
+            request.on('data', (data) => {
+                body += data;
+            });
+
+            request.on('end', () => {
+                let formFields: ?{[name: string]: string, ...};
+                try {
+                    formFields = JSON.parse(body);
+                } catch (e) {
+                    return sendResponse(response, headers, 500, 'Unsupported content-type ' + contentType);
+                }
+
+                const res = mockingServer.handle({
+                    method,
+                    urlPath: pathname || '',
+                    urlParams: query,
+                    headers,
+                    formFields,
+                });
+
+                if (res) {
+                    const {headers: responseHeaders, content, statusCode} = res;
+
+                    sendResponse(response, responseHeaders, statusCode, content);
+                }
+            });
```

Comment:
this part is also mostly the same

Reply:
I could reuse the stream reading part, end made the check agains the content type inside the 'end' callback